Simplenote/src/main/java/com/automattic/simplenote/DeepLinkActivity.java

```diff
@@ -0,0 +1,43 @@
+package com.automattic.simplenote;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.automattic.simplenote.utils.AuthUtils;
+
+import net.openid.appauth.RedirectUriReceiverActivity;
+
+import java.util.Locale;
+
+public class DeepLinkActivity extends AppCompatActivity {
+    private static final String AUTHENTICATION_SCHEME = "auth";
+    private static final String LOGIN_SCHEME = "login";
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Uri uri = getIntent().getData();
+        if (uri.getHost().equals(AUTHENTICATION_SCHEME)) {
+            Intent intent = new Intent(this, RedirectUriReceiverActivity.class);
+            intent.setData(uri);
+            startActivity(intent);
+        } else if (uri.getHost().equals(LOGIN_SCHEME)) {
+            Intent intent = new Intent(this, NotesActivity.class);
+            Simplenote app = (Simplenote) getApplication();
+            String email = AuthUtils.extractEmailFromMagicLink(uri);
+            if (app.isLoggedIn() &&
+                !email.toLowerCase(Locale.US).equals(app.getUserEmail().toLowerCase(Locale.US))) {
+                intent.putExtra(NotesActivity.KEY_ALREADY_LOGGED_IN, true);
+            } else {
+                AuthUtils.magicLinkLogin((Simplenote) getApplication(), uri);
```

Comment:
thanks for updating this. I can't leave a comment in the existing conversation, as all GitHub is showing me is a button to "resolve conversation," so I'm leaving this here.

I think this will be a valuable change to keep the login semantics separated, both in exposing `getUserEmail()` and also in adding `login` to `Simperium.java`. I don't understand the motivation behind taking a single semantic intention "is the user with this email logged in" and splitting it into separate internal details ("what is the email of the user" and "is the user logged in") (because we still end up duplicating logic from the main app in this activity right here) but this does seem better than it was and I don't want to hold up the PR on this point.

for curiosity, what was the purpose in extracting these internal details?

Reply:
We actually don't need to check "if the user with this email logged in", but it's "is the user logged in with a different email", which seems pretty coupled to the new signup flow, and not something reusable, and which should be handled by the Application class, so we decided to just expose the email of the currently logged in user, and keep the logic of the new signup flow contained in its place.

Reply:
> We actually don't need to check "if the user with this email is logged in", but it's "is the user logged in with a different email"

this took me a long time to understand and realize my mistake. thanks for clarifying the logic. it makes more sense now.